python_src/src/lamp_py/postgres/postgres_utils.py

```diff
@@ -380,16 +401,28 @@ def write_to_parquet(
         :param schema: schema of parquet file from select query
         :param batch_size: number of records to stream from db per batch
         """
+        process_logger = ProcessLogger(
+            "postgres_write_to_parquet",
+            batch_size=batch_size,
+            write_path=write_path,
+        )
+        process_logger.log_start()
+
+        part_stmt = select_query.execution_options(
+            stream_results=True,
+            max_row_buffer=batch_size,
+        )
         with self.session.begin() as cursor:
-            result = cursor.execute(select_query).yield_per(batch_size)
             with pq.ParquetWriter(write_path, schema=schema) as pq_writer:
-                for part in result.partitions():
+                for part in cursor.execute(part_stmt).partitions(batch_size):
```

Comment:
this is the real meat of the change, right?

any thoughts on why this was working before and stopped now?

Reply:
My intuition is that it was never really working as we expected. Hence the large spike in memory usage at each app start, where the LAMP_ALL_RT parquet file had to be fully created. But for some reason, that memory spike was being re-claimed at some point and I'm not sure why that stopped happening.